amd_openvx_extensions/amd_nn/nn_hip/nn_hip_kernels.cpp

```diff
@@ -31,6 +31,63 @@ typedef struct d_half4 {
   __half data[4];
 } d_half4;
 
+template <typename T>
+__global__ void __attribute__((visibility("default")))
+Hip_tensor_compare_layer(uchar* in, uint in_offset, uint4 in_stride, uchar* in2, uint in2_offset, uint4 in2_stride,
+ uchar* out, uint out_offset, uint4 out_stride, uint mode) {
+
+    uint x = hipBlockDim_x * hipBlockIdx_x + hipThreadIdx_x;
+    uint y = hipBlockDim_y * hipBlockIdx_y + hipThreadIdx_y;
+    uint z = hipBlockDim_z * hipBlockIdx_z + hipThreadIdx_z;
+
+    T value = *(T*)&in[in_offset + x * in_stride.x + y * in_stride.y + z * in_stride.z];
+    T value2 = *(T*)&in2[in2_offset + x * in2_stride.x + y * in2_stride.y + z * in2_stride.z];
+    out += out_offset + x * out_stride.x + y * out_stride.y + z * out_stride.z;
+   
+    // compare the values and write to the output\n"
+    bool result;
+    switch (mode) {
+        printf("the mode value is %d\n", mode);
```

Comment:
remove printf from here. The printf inside a kernel should only be used for debugging purposes.